test/common/typeconfig.go

```diff
@@ -21,13 +21,12 @@ import (
 
 	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
 	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	client "github.com/kubernetes-sigs/federation-v2/pkg/client/generic"
 )
 
 func GetTypeConfig(genericClient client.Client, name, namespace string) (typeconfig.Interface, error) {
 	typeConfig := &fedv1a1.FederatedTypeConfig{}
-	key := client.ObjectKey{Name: name, Namespace: namespace}
-	err := genericClient.Get(context.Background(), key, typeConfig)
+	err := genericClient.Get(context.Background(), typeConfig, namespace, name)
```

Comment:
(No action required) I'd still like to see the generic client methods avoid requiring a context (and expose other methods that do e.g. `Get` wouldn't require a context, `GetWithContext` would).

Reply:
I am still unable to decide whether it is a best practice to store the context in client structure as it is against the recommendation from the package authors. Here is one such discussion about it https://groups.google.com/forum/#!topic/golang-nuts/xRbzq8yzKWI

Reply:
My point wasn't to store a context, but rather to have a client call by default supplied a new context with each call since it's effectively just boilerplate.  Where we actually intend to do things like cancel we can call `<method>WithContext` or something.

Reply:
Ack